functions/src/export-csv.ts

```diff
@@ -167,3 +171,31 @@ function getFileName(jobName: string) {
   const fileBase = jobName.toLowerCase().replace(/[^a-z0-9]/gi, '-');
   return `${fileBase}.csv`;
 }
+
+function getAllPropertiesName(
+  lois: FirebaseFirestore.QuerySnapshot<FirebaseFirestore.DocumentData>): string[] {
+  let properties: string[] = [];
+  lois.forEach((loi) => {
+    Object.keys(loi.get('properties')).forEach((prop) => {
+      // We don't push 'id' because it's added initially in the first column of the CSV file.
+      if (!properties.includes(prop) && prop !== 'id') {
+        properties.push(prop);
+      }
+    })
+  })
+
+  return properties
+}
+
+function extractLoiProperties(
```

Comment:
Same idea would apply here, we'd probably return immutablejs `List`

Naming nit: `getPropertiesByName`.